adlfs/spec.py

```diff
@@ -2202,7 +2215,7 @@ async def _async_upload_chunk(self, final: bool = False, **kwargs):
                 # which is throws an InvalidHeader error from Azure, so instead
                 # of staging a block, we directly upload the empty blob
                 # This isn't actually tested, since Azureite behaves differently.
-                if block_id == self._get_block_id([]) and length == 0 and final:
+                if len(self._block_list) == 0 and length == 0 and final:
```

Comment:
Small nitpick, but we should just be able to do:
```python 
if not self._block_list and length == 0 and final:
```
It's a little bit shorter and more common to check if a list is falsey in order to determine if it is empty.